core/state-machine/src/ext.rs

```diff
@@ -329,6 +340,17 @@ where
 		self.changes_trie_transaction = root_and_tx;
 		root
 	}
+
+	fn submit_extrinsic(&mut self, extrinsic: Vec<u8>) -> Result<(), ()> {
+		let _guard = panic_handler::AbortGuard::new(true);
```

Comment:
I could possibly be wrong, but could you, please, verify @tomusdrw :
1) most of times runtime calls are performed with `NeverOffchainExt`;
2) `NeverOffchainExt::submit_extrinsic` is simply `unreachable!()`;
3) this guard means that if we met `panic!()`, `unreachable!()`, etc in the code, the process will `exit(1)` (see `panic_handler`);
4) the runtime could **try to** call `ext_submit_extrinsic` anytime (i.e. from `initialise_block`), and, given (1)..(3) the process will exit with 1 instead of failing execution?

Reply:
Oh, wait :) `NeverOffchainExt::new` returns `None`, so we will go to else branch here :)